snapshots/overlay/overlay.go

```diff
@@ -112,13 +129,31 @@ func NewSnapshotter(root string, opts ...Opt) (snapshots.Snapshotter, error) {
 		logrus.WithError(err).Warnf("cannot detect whether \"userxattr\" option needs to be used, assuming to be %v", userxattr)
 	}
 
+	metacopyOn := config.metacopyCtrl && config.metacopyOn && supportsMetacopy()
+	if metacopyOn && userxattr {
+		logrus.Warnf("userxattr, and metacopy=on can't be used together. Metacopy disabled.")
+		metacopyOn = false
+	}
+
+	if config.metacopyCtrl {
+		if metacopyOn {
```

Comment:
If we are keeping the two bools, then this is redundant. `metacopyOn` is true if `config.metacopyCtrl`.

Of course, you can't just change that for `if metacopyOn` as the else case would be broken (we can add when it is not supported). But it seems like this could be simplified.

Maybe the if should be: `if config.metacopyCtrl && supportsMetacopy()` and inside that you test for metacopyOn and userxattr?

Reply:
This is just debug leftover, I'll remove it, thanks!

Reply:
I think you introduced the bug I mentioned it is easy to introduce here. But let's move the conversation to: https://github.com/containerd/containerd/pull/7141/files#r929864915